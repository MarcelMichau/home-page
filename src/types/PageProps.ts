import { FixedObject } from 'gatsby-image';

export type PageProps = {
	data: {
		markdownRemark: {
			html: string;
		};
		file: {
			childImageSharp: {
				fixed: FixedObject[];
			};
		};
	};
};
