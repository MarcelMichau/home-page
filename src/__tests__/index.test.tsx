import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import IndexPage from '../pages/index';
import { PageProps } from '../types/PageProps';

const pageProps: PageProps = {
	data: {
		markdownContent: {
			html: `<p>This is my website. No one else's.</p>\n<p>Look how minimal it is.</p>\n<p>It's built using:</p>\n<ul>\n<li><a href=\"https://www.typescriptlang.org/\">TypeScript</a></li>\n<li><a href=\"https://reactjs.org/\">React</a></li>\n</ul>\n<p>Along with:</p>\n<ul>\n<li><a href=\"https://www.gatsbyjs.com/\">Gatsby</a></li>\n</ul>\n<p>With some help from:</p>\n<ul>\n<li><a href=\"https://styled-components.com/\">Styled Components</a></li>\n<li><a href=\"https://fontawesome.com/\">Font Awesome</a></li>\n<li><a href=\"https://getwaves.io/\">getwaves.io</a></li>\n</ul>\n<p>With code kept in line by:</p>\n<ul>\n<li><a href=\"https://codeclimate.com/github/MarcelMichau/home-page\">Code Climate</a></li>\n<li><a href=\"https://prettier.io/\">Prettier</a></li>\n</ul>\n<p>Tested using:</p>\n<ul>\n<li><a href=\"https://jestjs.io/\">Jest</a></li>\n<li><a href=\"https://www.docker.com/\">Docker</a></li>\n<li><a href=\"https://github.com/features/actions\">GitHub Actions</a></li>\n</ul>\n<p>Deployed to:</p>\n<ul>\n<li><a href=\"https://pages.cloudflare.com/\">Cloudflare Pages</a></li>\n<li><a href=\"https://workers.cloudflare.com/\">Cloudflare Workers</a></li>\n</ul>\n<p>And code hosted on:</p>\n<ul>\n<li><a href=\"https://github.com/MarcelMichau/home-page\">GitHub</a></li>\n</ul>\n<p>It is intended to be:</p>\n<ul>\n<li>Fast 🐌</li>\n<li>Agile 🐢</li>\n<li>Scalable ⚖️</li>\n<li>Cloud-native 🌧️</li>\n<li>Distributed 🎊</li>\n<li>Resilient 🔨</li>\n<li>Asynchronous ⏲</li>\n<li>Sustainable 🌲</li>\n<li>Privacy-focused 🔒</li>\n<li>Enterprise-ready 🏢</li>\n<li>Containerized 📦</li>\n<li>Built on Microservices 🔍</li>\n<li>Powered by Machine Learning 🤖</li>\n<li>Blockchain ₿</li>\n<li>Something about Artificial Intelligence 💡</li>\n<li>Gluten-free 🍕</li>\n<li>Filled with buzz words so that search engines pick this up 😁</li>\n</ul>`,
		},
	},
};

const renderIndexPageToJSON = () => {
	var json = renderer.create(<IndexPage data={pageProps.data} />).toJSON();
	return json;
};

describe('IndexPage', () => {
	it('renders correctly', () => {
		expect(renderIndexPageToJSON()).toMatchSnapshot();
	});
});
