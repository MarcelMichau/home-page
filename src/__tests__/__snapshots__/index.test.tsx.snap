// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
<div>
  <div>
    <div
      className=""
    >
      <header
        className=""
      >
        <div>
          <h2
            className=""
          >
            Marcel Michau
          </h2>
          <div
            className=""
          >
            <span
              className=""
            >
              <span
                className=""
              >
                console
              </span>
              <span
                className=""
              >
                .
              </span>
              <span
                className=""
              >
                log
              </span>
              (
              <span
                className=""
              >
                <span
                  className=""
                >
                  "
                </span>
                I write code & stuff
                <span
                  className=""
                >
                  "
                </span>
              </span>
              )
              <span
                className=""
              >
                ;
              </span>
            </span>
          </div>
          <div
            className=""
          >
            <a
              aria-label="GitHub Profile"
              href="https://github.com/MarcelMichau"
              title="GitHub Profile"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-github fa-2x "
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </a>
            <a
              aria-label="DEV Community Profile"
              href="https://dev.to/marcelmichau"
              title="DEV Community Profile"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-dev fa-2x "
                data-icon="dev"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M120.12 208.29c-3.88-2.9-7.77-4.35-11.65-4.35H91.03v104.47h17.45c3.88 0 7.77-1.45 11.65-4.35 3.88-2.9 5.82-7.25 5.82-13.06v-69.65c-.01-5.8-1.96-10.16-5.83-13.06zM404.1 32H43.9C19.7 32 .06 51.59 0 75.8v360.4C.06 460.41 19.7 480 43.9 480h360.2c24.21 0 43.84-19.59 43.9-43.8V75.8c-.06-24.21-19.7-43.8-43.9-43.8zM154.2 291.19c0 18.81-11.61 47.31-48.36 47.25h-46.4V172.98h47.38c35.44 0 47.36 28.46 47.37 47.28l.01 70.93zm100.68-88.66H201.6v38.42h32.57v29.57H201.6v38.41h53.29v29.57h-62.18c-11.16.29-20.44-8.53-20.72-19.69V193.7c-.27-11.15 8.56-20.41 19.71-20.69h63.19l-.01 29.52zm103.64 115.29c-13.2 30.75-36.85 24.63-47.44 0l-38.53-144.8h32.57l29.71 113.72 29.57-113.72h32.58l-38.46 144.8z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </a>
            <a
              aria-label="Twitter Profile"
              href="https://twitter.com/MarcelMichau"
              title="Twitter Profile"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter fa-2x "
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </a>
          </div>
        </div>
      </header>
      <main
        className=""
      >
        <div
          dangerouslySetInnerHTML={
            {
              "__html": "<p>This is my website. No one else's.</p>
<p>Look how minimal it is.</p>
<p>It's built using:</p>
<ul>
<li><a href="https://www.typescriptlang.org/">TypeScript</a></li>
<li><a href="https://reactjs.org/">React</a></li>
</ul>
<p>Along with:</p>
<ul>
<li><a href="https://www.gatsbyjs.com/">Gatsby</a></li>
</ul>
<p>With some help from:</p>
<ul>
<li><a href="https://styled-components.com/">Styled Components</a></li>
<li><a href="https://fontawesome.com/">Font Awesome</a></li>
<li><a href="https://getwaves.io/">getwaves.io</a></li>
</ul>
<p>With code kept in line by:</p>
<ul>
<li><a href="https://codeclimate.com/github/MarcelMichau/home-page">Code Climate</a></li>
<li><a href="https://prettier.io/">Prettier</a></li>
</ul>
<p>Tested using:</p>
<ul>
<li><a href="https://jestjs.io/">Jest</a></li>
<li><a href="https://www.docker.com/">Docker</a></li>
<li><a href="https://github.com/features/actions">GitHub Actions</a></li>
</ul>
<p>Deployed to:</p>
<ul>
<li><a href="https://pages.cloudflare.com/">Cloudflare Pages</a></li>
</ul>
<p>And code hosted on:</p>
<ul>
<li><a href="https://github.com/MarcelMichau/home-page">GitHub</a></li>
</ul>
<p>It is intended to be:</p>
<ul>
<li>Fast 🐌</li>
<li>Agile 🐢</li>
<li>Scalable ⚖️</li>
<li>Cloud-native 🌧️</li>
<li>Distributed 🎊</li>
<li>Resilient 🔨</li>
<li>Asynchronous ⏲</li>
<li>Sustainable 🌲</li>
<li>Privacy-focused 🔒</li>
<li>Enterprise-ready 🏢</li>
<li>Containerized 📦</li>
<li>Built on Microservices 🔍</li>
<li>Powered by Machine Learning 🤖</li>
<li>Blockchain ₿</li>
<li>Something about Artificial Intelligence 💡</li>
<li>Gluten-free 🍕</li>
<li>Filled with buzz words so that search engines pick this up 😁</li>
</ul>",
            }
          }
        />
      </main>
      <svg
        className=""
        viewBox="0 0 1440 320"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0,192L48,192C96,192,192,192,288,160C384,128,480,64,576,80C672,96,768,192,864,208C960,224,1056,160,1152,154.7C1248,149,1344,203,1392,229.3L1440,256L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
          fill="#400202"
          fillOpacity="1"
        />
      </svg>
      <div>
        <footer
          className=""
        >
          <div>
            Marcel Michau © 
            2023
          </div>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
